server/src/infra/repositories/person.repository.ts

```diff
@@ -96,14 +96,32 @@ export class PersonRepository implements IPersonRepository {
     return this.personRepository.findOne({ where: { id: personId } });
   }
 
-  getByName(userId: string, personName: string): Promise<PersonEntity[]> {
-    return this.personRepository
+  getByName(userId: string, personName: string, withHidden: boolean | undefined): Promise<PersonEntity[]> {
+    const queryBuilder = this.personRepository
       .createQueryBuilder('person')
       .leftJoin('person.faces', 'face')
       .where('person.ownerId = :userId', { userId })
-      .andWhere('LOWER(person.name) LIKE :name', { name: `${personName.toLowerCase()}%` })
-      .limit(20)
-      .getMany();
+      .andWhere('LOWER(person.name) LIKE :name', { name: `%${personName.toLowerCase()}%` })
+      .groupBy('person.id')
+      .orderBy('COUNT(face.assetId)', 'DESC')
+      .limit(20);
+
+    if (!withHidden) {
+      queryBuilder.andWhere('person.isHidden = false');
+    }
+    return queryBuilder.getMany();
+  }
+
+  getPersonAssetsCount(personId: string): Promise<number> {
+    return this.assetFaceRepository
+      .createQueryBuilder('face')
+      .leftJoin('face.asset', 'asset')
+      .where('face.personId = :personId', { personId })
+      .andWhere('asset.isArchived = false')
+      .andWhere('asset.deletedAt IS NULL')
+      .andWhere('asset.livePhotoVideoId IS NULL')
+      .distinct(true)
+      .getCount();
```

Comment:
This is in the person repository, can we just call it `getAssetCount(personId: string)` instead? If we rename this stuff to statistics, we can use `getStatistics` instead.